daemon/daemon.go

```diff
@@ -1451,6 +1452,15 @@ func (daemon *Daemon) verifyContainerSettings(hostConfig *containertypes.HostCon
 				return nil, err
 			}
 		}
+
+		// Validate if the given hostname is RFC 1123 (https://tools.ietf.org/html/rfc1123) compliant.
+		if len(config.Hostname) > 0 {
+			// Linux hostname is limited to HOST_NAME_MAX=64, not not including the terminating null byte.
```

Comment:
From a NetBIOS perspective, it's 15. But unfortunately not that clear cut...

![image](https://cloud.githubusercontent.com/assets/10522484/14149896/3c55eb4e-f65b-11e5-829f-796db501bd11.png)

From an actual computername, it's 63 - You can't type any more characters after that.

![image](https://cloud.githubusercontent.com/assets/10522484/14149972/9f21c39c-f65b-11e5-836f-0b2d700abeda.png)

![image](https://cloud.githubusercontent.com/assets/10522484/14150043/fc728ea0-f65b-11e5-80a7-66615d64c45e.png)


So the absolute limit can't be more than 63, but I'm not sure whether we should actually limit it to 15 on Windows. @jstarks do you have an opinion here?